File: strings/largestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        st=""

        first=strs[0]
        last=strs[-1]

        for i in range(min(len(first), len(last))):
            if first[i]!=last[i]:
                break

            st+=first[i]

        return st

File: strings/largestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string st = "";
        int n=strs.size();
        sort(strs.begin(), strs.end());

        string first = strs[0];
        string last = strs[n-1];

        int minL=min(first.size(), last.size());

        for(int i=0; i<minL; i++){
            if(first[i]!=last[i]) break;
            st.push_back(first[i]);
        }
        return st;
    }
};